pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -455,38 +478,113 @@ func (r *Reconciler) getOrCreateClusterResourceSnapshot(ctx context.Context, crp
 		return nil, err
 	}
 
-	// create a new resource snapshot
-	latestResourceSnapshotIndex++
-	latestResourceSnapshot = &fleetv1beta1.ClusterResourceSnapshot{
+	// only increment if new cluster resource snapshots are to be created.
```

Comment:
The solution should work for the positive case, but the perf is not good for the negative case when the controller crashes in the middle. And there's a bug when handling the redundant resource snapshots.